vars/parseStageInfo.groovy

```diff
@@ -144,6 +144,10 @@ def call(Map config = [:]) {
         result['pragma_suffix'] = '-hw-medium'
       }
     }
+    if (stage_name.contains('with Valgrind')) {
```

Comment:
Putting this here feels like it would have (too) broad coverage.  If somebody were to add some other stage that simply had "with Valgrind" in it, these overrides would affect that stage also.  I think you want to target this check to a more specific stage name, possibly using the block above https://github.com/daos-stack/pipeline-lib/blob/f5cb9cc8a6561b2f871ecf7de209611f176c8b7c/vars/parseStageInfo.groovy#L127-L146

Reply:
I moved it into the above block, so we set `WITH_VALGRIND=memcheck` for a stage name match on `Functional` and `with Valgrind`.